### PR TITLE
Server building and improovement

### PR DESCRIPTION
**Pull request by galyfray**
- added build function for the server
  - refactored existing building script
  - added npm commends
    - `build-web` - only build the web part of the app
    - `build-serv` - only build the server part of the app
    - `start-serv` - `build the server and then start it`
- made the server send the web files
- fixed few bugs
- centralized eslint config file

### CODE REVIEW

File: Scripts/devEnv.js [hunk 1]
```diff
@@ -44,83 +44,136 @@ async function getAllFiles(dir) {
...
             })
         ) {
             BUFFER.push("[ERROR][LINT] Linting failed.");
             console.log("[ERROR][LINT] Linter failed to fix all problem, see log file for details.");
             throw new Error("Linting failed");
+        } else {
+            BUFFER.push("[INFO][LINT] Linting successful.");
         }
+    } catch (e) {
+        BUFFER.push("[ERROR][LINT] Linting failed due to an unexpected error");
```
Comment (Yasmarila) [inline Scripts/devEnv.js]: put a dot

File: Scripts/devEnv.js [hunk 2]
```diff
@@ -133,10 +186,30 @@ async function getAllFiles(dir) {
 
         if (errorFlag) {
             throw new Error("Compilation failed");
+        } else {
+            BUFFER.push("[INFO][AS] Compilation successful.");
         }
+    } catch (e) {
+        BUFFER.push("[ERROR][AS] Comilation failed due to an unexpected error");
```
Comment (Yasmarila) [inline Scripts/devEnv.js]: Comilation → Compilation
put a dot ;)

File: Scripts/devEnv.js [hunk 3]
```diff
@@ -168,62 +241,227 @@ async function getAllFiles(dir) {
...
-            pending.push(
-                fs.promises.copyFile(
-                    path.join(process.argv[4], source),
-                    path.join(DEV_DIR, source)
-                )
-            );
-        });
+/**
+ * This function will copy the given files to their destination.
+ * Their destination is define by the TRANSFORM function.
```
Comment (Yasmarila) [inline Scripts/devEnv.js]: define → defined

File: Scripts/devEnv.js [hunk 4]
```diff
@@ -168,62 +241,227 @@ async function getAllFiles(dir) {
...
+            await fs.promises.mkdir(path.dirname(filename), {recursive: true}).then(() => {
+                return fs.promises.copyFile(
+                    file,
+                    filename
+                );
+            });
+        }));
+        BUFFER.push("[INFO][COPY] Copying successful.");
+    } catch (e) {
+        BUFFER.push("[ERROR][COPY] Faled to copy all the files");
```
Comment (Yasmarila) [inline Scripts/devEnv.js]: Faled → Failed

**Review by Yasmarila — CHANGES_REQUESTED** [review]

File: Scripts/devEnv.js (hunk 1 above)
Comment (galyfray) [reply Scripts/devEnv.js]: sir! yes, sir!

**Review by Yasmarila — APPROVED** [review]

**Pull request merged**